### ISSUE TITLE
"Amazing Tunes" does not load

### ISSUE THREAD

**Issue by dirkwhoffmann** [dirkwhoffmann/vAmiga#689]
Reported by @chris70c: 
[AmazingTunes.zip](https://github.com/dirkwhoffmann/vAmiga/files/8586095/AmazingTunes.zip)

<img width="1044" alt="Bildschirmfoto 2022-04-28 um 22 11 28" src="https://user-images.githubusercontent.com/12561945/165839023-0fae0b27-3422-41ab-9d12-bf171439c03e.png">

It loads fine in UAE.

TODO: Find and try the ADF version of the demo.

**Comment by mras0** [comment]
Are you using the same config in WinUAE as you are in vAmiga? I get the same message in WinUAE if I add some fast ram to the "most compatible" configuration (and use cycle exact), and it appears to work in vAmiga in classic 512K chip + 512 slow config.

**Comment by chris70c** [comment]
Yes, exactly the same config:
KS 2.04, Quickstart A500+ then 2MB Chip, 8MB Fast, add .hdf file.
Like I told to Dirk, this doesn't happens with the .adf file is something related to whdload, there are several install that reports error with autovector and they work fine on WinUAE and/or real Amiga hardware.
If you want I can send you the .hdf file...

**Comment by mras0** [comment]
@chris70c When you're testing with WinUAE do you have the compatibility slider all the way to the left (this isn't default)? Like this:

![image](https://user-images.githubusercontent.com/9952017/166119037-39aeece0-ddb4-458b-a133-e99a57291ff0.png)

This is the config I'm using, A500+ quickstart (but slider adjusted) + 2MB chip + 8MB fast (HDF is from first post in the thread).

[AmazingTune-uae-configs.zip](https://github.com/dirkwhoffmann/vAmiga/files/8597631/AmazingTune-uae-configs.zip)

What config do you test on a real Amiga? It does work on my 060, but I think 7MHz 68000 + fast ram is quite an uncommon configuration, so maybe it wasn't tested? Like #678 everything seems to work if I up the CPU speed to 14Mhz in WinUAE (probably more common).

**Comment by chris70c** [comment]
using your config:
![Untitled](https://user-images.githubusercontent.com/3858768/166120473-015c6819-2cf2-4df9-a96d-f2a57540a2b1.png)

**Comment by mras0** [comment]
WinUAE could still be wrong of course, so it might be interesting if somebody could test one of these whdload slaves on the real thing. Trapdoor fast ram expansions seem to be available, and if I'm reading the specs right maybe something like a ACA500 could be configured to use the normal CPU but still provide true fast ram.

DMS version of the music disk can be downloaded via the [whdload page](http://www.whdload.de/demos/A.html), but I've also attached it here: [SAE-AmazingTunes.zip](https://github.com/dirkwhoffmann/vAmiga/files/8598513/SAE-AmazingTunes.zip).

**Comment by dirkwhoffmann** [comment]
I've got an ACA500+, but I have no idea how to test whdload related stuff with it. I have a fleet of A500s around, but no machine with hard drive capabilities. I would need an A600, wouldn't I?

**Comment by mras0** [comment]
If you have an extra floppy drive connected, you should be able to test it that way. It's small enough that the slave fits on one disk, unfortunately it's just too large to also fit the WHDload executable and disk swapping is not supported.

[WFDLoad-AmazingTunes.zip](https://github.com/dirkwhoffmann/vAmiga/files/8598855/WFDLoad-AmazingTunes.zip)

**Comment by dirkwhoffmann** [comment]
I found this: 

![gotek](https://user-images.githubusercontent.com/12561945/166146293-fe9cb4ff-2c9f-4723-9663-f1cd093f00f4.jpg)

I'll try to see if I can assemble a second drive with it (need to read the instructions).

**Comment by dirkwhoffmann** [comment]
Got the second drive working 😎:

![IMG_2713](https://user-images.githubusercontent.com/12561945/166147663-91291117-c475-43db-b155-ca60f0fafe1c.JPG)

When booting with "AmazingTunes.adf" in df0: and "WhdLoad.adf" in df1:, I get an error message (same in vAmiga):

![IMG_2712](https://user-images.githubusercontent.com/12561945/166147702-58782bd1-2d17-4c53-8fdc-f85ce91983ad.JPG)

What is the expected outcome? BTW, this is not the ACA500+ machine. I wanted to get the second drive going before switching to modded machines.

**Comment by mras0** [comment]
That's the expected outcome in KS1.x (at least I see the same). It should work with KS2.0+ (I edited my old post, ACA+ should include KS3.1 as far as I can tell).

**Comment by dirkwhoffmann** [comment]
> That's the expected outcome in KS1.x

Ok, cool, let's move on to the next level:

![IMG_2714](https://user-images.githubusercontent.com/12561945/166149442-67de4634-0433-40d7-9d3d-6fa610f85b03.JPG)

This is the startup config (Kick 3.1, FastRam enabled):

![IMG_2719](https://user-images.githubusercontent.com/12561945/166149473-e0789ccb-664d-4de7-a385-59123a3b7472.JPG)

And this is the result:

![IMG_2720](https://user-images.githubusercontent.com/12561945/166149492-6182165d-7afa-4c48-b4cb-1424fd1849e3.JPG)

Bottom line: Not a bug. Our emulators are doing the right thing.

**Comment by mras0** [comment]
Super cool, and great you had all the necessary hardware at hand!

BTW isn't the part labelled "boot card" on the ACA a CF-card reader that should allow you to test HD stuff?

**Comment by dirkwhoffmann** [comment]
> BTW isn't the part labelled "boot card" on the ACA a CF-card reader that should allow you to test HD stuff?

Yes, you're right. I think it's supposed to work that way. Up to now, I only used it once to install a custom Rom (which failed for whatever reason).
